external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -153,11 +201,7 @@ private[kafka010] case class KafkaSource(
 
```

Comment:
`currentPartitionOffsets` should be set to `untilPartitionOffsets` if it's empty. It means recovery from a failure.